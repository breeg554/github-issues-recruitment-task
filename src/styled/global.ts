import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  html,button,input,textarea{
    font-family: 'Roboto', sans-serif;
    font-size: 10px;   //1rem = 10px
  }
  body{
    background-color: #E5E5E5;
  }
`;
