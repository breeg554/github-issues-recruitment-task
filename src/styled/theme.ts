export type ThemeType = typeof theme; // This is the type definition for my theme object.

export const theme = {
  colors: {
    core: {
      primary: "#24292F",
      secondary: "#166CD7",
      tertiary: "#6F7781",
    },
    hover: {
      primary: "#353a3F",
      secondary: "#288ef9",
      tertiary: "#9F9992",
    },
  },
  mediaQ: {
    sm: "@media(min-width: 576px)",
    md: "@media(min-width: 768px)",
    lg: "@media(min-width: 992px)",
    xl: "@media(min-width: 1200px)",
  },
};
