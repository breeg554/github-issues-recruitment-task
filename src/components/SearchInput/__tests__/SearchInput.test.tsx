import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../lib/styled/theme";
import { SearchInput } from "..";
describe("Search input component", () => {
  it("renders out and match the snapshot", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <SearchInput value="abc" onChange={() => null} />
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
