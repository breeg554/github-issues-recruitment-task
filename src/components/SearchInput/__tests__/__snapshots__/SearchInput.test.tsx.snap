// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search input component renders out and match the snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL kPkscH"
  >
    <input
      aria-label="Search users and repositories"
      class="sc-gsDKAQ jAXgbX search-input"
      placeholder="Search"
      value="abc"
    />
  </div>
</div>
`;
